modules/dev-tools/deprecation.php

```diff
@@ -0,0 +1,190 @@
+<?php
+namespace Elementor\Modules\DevTools;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Deprecation {
+	const SOFT_VERSIONS_COUNT = 4;
+	const HARD_VERSIONS_COUNT = 8;
+
+	private $current_version = null;
+	private $soft_deprecated_notices = [];
+
+	public function __construct( $current_version ) {
+		$this->current_version = $current_version;
+	}
+
+	public function get_settings() {
+		return [
+			'soft_notices' => $this->soft_deprecated_notices,
+		];
+	}
+
+	/**
+	 * Get total of major.
+	 *
+	 * Since `get_total_major` cannot determine how much really versions between 2.9.0 and 3.3.0 if there is 2.10.0 version for example,
+	 * versions with major2 more then 9 will be added to total.
+	 *
+	 * @param array $parsed_version
+	 *
+	 * @return int
+	 */
+	public function get_total_major( $parsed_version ) {
+		$major1 = $parsed_version['major1'];
+		$major2 = $parsed_version['major2'];
+		$major2 = $major2 > 9 ? 9 : $major2;
+		$minor = 0;
+
+		$total = intval( "{$major1}{$major2}{$minor}" );
+
+		if ( $total > 99 ) {
+			$total = $total / 10;
+		} else {
+			$total = intval( $total / 10 );
+		}
+
+		if ( $parsed_version['major2'] > 9 ) {
+			$total += $parsed_version['major2'] - 9;
+		}
+
+		return $total;
+	}
+
+	/**
+	 * Get's next version.
+	 *
+	 * @param string $version
+	 * @param int $count
+	 *
+	 * @return string|false
+	 */
+	public function get_next_version( $version, $count = 1 ) {
+		$version = $this->parse_version( $version );
+
+		if ( ! $version ) {
+			return false;
+		}
+
+		$version['total'] = $this->get_total_major( $version ) + $count;
+
+		$total = $version['total'];
+
+		if ( $total > 9 ) {
+			$version['major1'] = intval( $total / 10 );
+			$version['major2'] = $total % 10;
+		} else {
+			$version['major1'] = 0;
+			$version['major2'] = $total;
+		}
+
+		$version['minor'] = 0;
+
+		return $this->implode_version( $version );
+	}
+
+	/**
+	 * Implode parsed version to string version.
+	 *
+	 * @param object $parsed_version
+	 *
+	 * @return string
+	 */
+	public function implode_version( $parsed_version ) {
+		$major1 = $parsed_version['major1'];
+		$major2 = $parsed_version['major2'];
+		$minor = $parsed_version['minor'];
+
+		return "{$major1}.{$major2}.{$minor}";
+	}
+
+	/**
+	 * Parse version('0.0.0') to an informative array.
+	 *
+	 * @param string $version
+	 *
+	 * @return array|false
+	 */
+	public function parse_version( $version ) {
+		$version_explode = explode( '.', $version );
+
+		if ( count( $version_explode ) != 3 ) {
```

Comment:
If the original version is something like `1.0.2.8` we need to drop the rest of the numbers before this check

Reply:
Done.